js/modules/k6/grpc/client.go

```diff
@@ -0,0 +1,575 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2020 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package grpc
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dop251/goja"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/protoparse"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	grpcstats "google.golang.org/grpc/stats"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"google.golang.org/protobuf/types/dynamicpb"
+
+	//nolint: staticcheck
+	protoV1 "github.com/golang/protobuf/proto"
+
+	"github.com/loadimpact/k6/js/common"
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/lib/metrics"
+	"github.com/loadimpact/k6/stats"
+)
+
+//nolint: lll
+var (
+	errInvokeRPCInInitContext = common.NewInitContextError("invoking RPC methods in the init context is not supported")
+	errConnectInInitContext   = common.NewInitContextError("connecting to a gRPC server in the init context is not supported")
+)
+
+// Client reprecents a gRPC client that can be used to make RPC requests
+type Client struct {
+	mds  map[string]protoreflect.MethodDescriptor
+	conn *grpc.ClientConn
+}
+
+// NewClient creates a new gPRC client to make invoke RPC methods.
+func (*GRPC) NewClient(ctxPtr *context.Context) interface{} {
+	rt := common.GetRuntime(*ctxPtr)
+
+	return common.Bind(rt, &Client{}, ctxPtr)
+}
+
+// MethodInfo holds information on any parsed method descriptors that can be used by the goja VM
+type MethodInfo struct {
+	grpc.MethodInfo
```

Comment:
```suggestion
	grpc.MethodInfo `json:"-" js:"-"`
```

I think I was a bit overzealous with the suggestion for code reuse here :disappointed: Without this change, if you run something like:
```js
let methods = client.load([], "samples/grpc_server/route_guide.proto");
console.log(JSON.stringify(methods, null, 4));
```
you'd get this:
```js
[
    {
        "method_info": {
            "name": "GetFeature",
            "is_client_stream": false,
            "is_server_stream": false
        },
        "name": "GetFeature",
        "is_client_stream": false,
        "is_server_stream": false,
        "package": "main",
        "service": "RouteGuide",
        "full_method": "/main.RouteGuide/GetFeature"
    },
    {
        "method_info": {
            "name": "ListFeatures",
            "is_client_stream": false,
            "is_server_stream": true
        },
        "name": "ListFeatures",
        "is_client_stream": false,
        "is_server_stream": true,
        "package": "main",
        "service": "RouteGuide",
        "full_method": "/main.RouteGuide/ListFeatures"
    },
    // ...
]
```

but yeah, I'm not sure if the anonymous struct reuse is worth it at this point :man_shrugging: